uefi/src/proto/device_path/mod.rs

```diff
@@ -119,6 +119,20 @@ pub struct DevicePathHeader {
     pub length: u16,
 }
 
+impl DevicePathHeader {
+    /// Create a [`DevicePathHeader`] reference from an opaque pointer.
+    ///
+    /// # Safety
+    ///
+    /// The input pointer must point to valid data. That data must
+    /// remain valid for the lifetime `'a`, and cannot be mutated during
+    /// that lifetime.
+    #[must_use]
+    pub unsafe fn from_ffi_ptr<'a>(ptr: *const FfiDevicePath) -> &'a DevicePathHeader {
```

Comment:
Is an `unsafe` interface needed here? How about another `TryFrom<&[u8]>` impl instead -- then it can check the number of bytes to ensure that the input is valid.

Reply:
sgtm